src/pages/quiz/QuizVariants.tsx

```diff
@@ -122,14 +120,17 @@ const ActiveQuiz: React.FC<ActiveQuizProps> = ({ finishQuiz }) => {
     setIsOpen(false)
   }, [])
 
-  const handleFinish = useCallback(async () => {
+  const handleFinish = async () => {
     const finishedQuiz = await mutateAsync()
-    setIsOpen(false)
-    finishQuiz(finishedQuiz?._id)
-    if (!scoredResponses) {
-      navigate(-1)
+    if (scoredResponses) {
```

Comment:
Always add empty lines around `if` statements. It will improve code readability a lot.
```suggestion
   
    if (scoredResponses) {
```

Reply:
fixed